lib/bolt/shell/powershell.rb

```diff
@@ -167,6 +167,25 @@ def with_tmpdir
         end
       end
 
+      private def restore_ruby_vars(options: {})
+        # Only do this for the local transport, not if bundled-ruby is nil
+        if target.transport_config['bundled-ruby'] == false
+          ruby_env_vars = RUBY_ENV_VARS.each_with_object({}) do |e, acc|
+            # Bundler sets GEM_PATH, GEM_HOME, RUBYLIB, and RUBYOPT
+            if ENV["BUNDLER_ORIG_#{e}"] &&
+               ENV["BUNDLER_ORIG_#{e}"] != "BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL"
+              acc[e] ||= ENV["BUNDLER_ORIG_#{e}"]
+            end
+            acc[e] ||= ""
+          end
+          options[:env_vars] ||= {}
+          # Let any user-defined env vars override our attempts to restore the
+          # Ruby environment
+          options[:env_vars] = ruby_env_vars.merge(options[:env_vars])
+        end
+        options
+      end
```

Comment:
Can this be a method on the base `Bolt::Shell` class so it can be reused? It looks like the bash shell uses `options[:environment]` over `options[:env_vars]` so that'd need to be updated, too.

Reply:
Hmm, I'd honestly rather just remove this from PowerShell altogether. We don't unset these env vars in the MSI, only in *nix, so this is only for unsetting the bundled environment which I just realized should already be covered by `Bundler.without_bundled` in the local transport.